docs/source/l1_to_l2b_e2e.rst

```diff
@@ -0,0 +1,198 @@
+
+
+Running the Test
+~~~~~~~~~~~~~~~~
+
+There are several ways to run the L1 to L2b end-to-end test:
+
+1. Using pytest
+   
+   .. code-block:: bash
+
+      # From the root directory of corgidrp
+      pytest tests/test_l1_to_l2b.py -v
+
+2. Direct execution
+   
+   .. code-block:: bash
+
+      # Run the script directly with default paths
+      python tests/test_l1_to_l2b.py
+
+      # Or specify custom paths
+      python tests/test_l1_to_l2b.py --tvacdata_dir /path/to/CGI_TVAC_Data --outputdir /path/to/output
+
+
+Test Setup
+~~~~~~~~~~
+
+To run this test, you will need three specific datasets from the database:
+
+1. **L1 Data** - Raw Level 1 data files.
+2. **L2b Data** - Processed Level 2b data files.
+3. **Cals** - Calibration files required for processing.
+
+If you don't have all the datasets downloaded, make sure you at least have these three files to run the test promptly. The script imports necessary libraries and determines the directory containing the script.
+   
+.. code-block:: python
+
+    import argparse
+    import os
+    import pytest
+    import numpy as np
+    import astropy.time as time
+    import astropy.io.fits as fits
+    import corgidrp
+    import corgidrp.data as data
+    import corgidrp.mocks as mocks
+    import corgidrp.walker as walker
+    import corgidrp.caldb as caldb
+    import corgidrp.detector as detector
+
+    thisfile_dir = os.path.dirname(__file__)  # this file's folder
+
+The `test_l1_to_l2b` function initializes paths for input data, calibration files, and output directories.
+
+.. code-block:: python
+
+    @pytest.mark.e2e
+    def test_l1_to_l2b(tvacdata_path, e2eoutput_path):
+        # Define input data paths
+        l1_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L1")
+        l2b_datadir = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "L2b")
+        processed_cal_path = os.path.join(tvacdata_path, "TV-36_Coronagraphic_Data", "Cals")
+
+        # Create output directory if it does not exist
+        l2b_outputdir = os.path.join(e2eoutput_path, "l1_to_l2b_output")
+        if not os.path.exists(l2b_outputdir):
+            os.mkdir(l2b_outputdir)
+
+Calibration files such as non-linearity tables, dark current, flat fields, and bad pixel maps are loaded.
+
+.. code-block:: python
+
+        nonlin_path = os.path.join(processed_cal_path, "nonlin_table_240322.txt")
+        dark_path = os.path.join(processed_cal_path, "dark_current_20240322.fits")
+        flat_path = os.path.join(processed_cal_path, "flat.fits")
+        fpn_path = os.path.join(processed_cal_path, "fpn_20240322.fits")
+        cic_path = os.path.join(processed_cal_path, "cic_20240322.fits")
+        bp_path = os.path.join(processed_cal_path, "bad_pix.fits")
+
+The raw science data files and mock calibration files are defined.
```

Comment:
this block of code should be organized with the input data block at the top. And note that the filename convention will be totally different for the real data, so that should be recognized.